output.c

```diff
@@ -143,17 +219,11 @@ output_destroy(struct cg_output *output)
 
 	free(output);
 
-	if (wl_list_empty(&server->outputs)) {
-		wl_display_terminate(server->wl_display);
-	} else if (server->output_mode == CAGE_MULTI_OUTPUT_MODE_LAST) {
+	if (server->output_mode == CAGE_MULTI_OUTPUT_MODE_LAST) {
 		struct cg_output *prev = wl_container_of(server->outputs.next, prev, link);
```

Comment:
@emersion, @Hjdskes 
I have tested `-m last` option to check modifications done in `output_disable()` and have observed several segfaults.
Reading the code and checking `wl_list` API, I feel that this is not the right way to reliably retrieve the next item in list. It seems `next` could never be `NULL` and the code can consequently try to release incorrect part of the memory.

By the way, is the `-m` option still relevant if wlr-output-managent support is added?

Reply:
Yeah, it's not safe to use `wl_container_of()` without checking first that the pointer passed to it is `!= &server->outputs`.